File: october_2024/gfg_potd_18Oct_2024.java
import java.util.*;

class Solution {
    public int getSingle(int[] arr) {
        // Using XOR to find the number occurring an odd number of times
        int result = 0;
        for (int num : arr) {
            result ^= num;
        }
        return result;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = Integer.parseInt(scanner.nextLine());
        
        while (t-- > 0) {
            String input = scanner.nextLine();
            String[] parts = input.split(" ");
            int[] arr = new int[parts.length];
            
            for (int i = 0; i < parts.length; i++) {
                arr[i] = Integer.parseInt(parts[i]);
            }

            Solution obj = new Solution();
            int res = obj.getSingle(arr);
            System.out.println(res);
        }
        
        scanner.close();
    }
}

File: october_2024/gfg_potd_18Oct_2024.cpp
/*
        Given an array arr[] of positive integers where every element appears even times except for one.
        Find that number occurring an odd number of times.

Examples:

Input: arr[] = [1, 1, 2, 2, 2]
Output: 2
Explanation: In the given array all element appear two times except 2 which appears thrice.
*/

//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;

// } Driver Code Ends
// User function template for C++
class Solution
{
public:
    int getSingle(vector<int> &arr)
    {
        // we will use xor operator to solve this problem efficiently.
        //  as, x^0=x. and x^x=0
        // start to iterate over the array, & at last
        // we will get our required answer.
        int a = 0;
        for (int i = 0; i < arr.size(); i++)
        {
            a = a ^ arr[i];
        }
        return a;
    }
};

//{ Driver Code Starts.

int main()
{
    string ts;
    getline(cin, ts);
    int t = stoi(ts);
    while (t--)
    {

        vector<int> arr;
        string input;
        getline(cin, input);
        stringstream ss(input);
        int number;
        while (ss >> number)
        {
            arr.push_back(number);
        }
        Solution obj;
        int res = obj.getSingle(arr);
        cout << res << endl;
    }
    return 0;
}
// } Driver Code Ends